pkg/schema/loader.go

```diff
@@ -110,84 +108,88 @@ func (r *Loader) del(key string) {
 
 // GetRef return an error if the ref does not exist
 // If the ref exists the ref is retrieved with an indication if the base provider schema version dir exists
-func (r *Loader) GetRef(ctx context.Context, key string) (*invv1alpha1.SchemaSpec, bool, error) {
-	ref, exists := r.get(key)
+func (r *Loader) GetRef(ctx context.Context, key string) (*invv1alpha1.Schema, bool, error) {
+	schema, exists := r.get(key)
 	if !exists {
 		return nil, false, fmt.Errorf("no repository reference registered for key %q", key)
 	}
-	baseRefPath := ref.GetBasePath(r.schemaDir)
+	baseRefPath := schema.Spec.GetBasePath(r.schemaDir)
 
-	return ref, utils.DirExists(baseRefPath), nil
+	return schema, utils.DirExists(baseRefPath), nil
 }
 
-func (r *Loader) get(key string) (*invv1alpha1.SchemaSpec, bool) {
+func (r *Loader) get(key string) (*invv1alpha1.Schema, bool) {
 	r.m.RLock()
 	defer r.m.RUnlock()
-	ref, exists := r.schemas[key]
-	return ref, exists
+	schema, exists := r.schemas[key]
+	return schema, exists
 }
 
-func (r *Loader) Load(ctx context.Context, key string, credentialNSN types.NamespacedName) error {
-	logger := log.FromContext(ctx)
+func (r *Loader) Load(ctx context.Context, key string) error {
+	log := log.FromContext(ctx)
 
-	schemaSpec, _, err := r.GetRef(ctx, key)
+	schema, _, err := r.GetRef(ctx, key)
 	if err != nil {
 		return err
 	}
 
-	var downloader downloadable
+	
 
 	// for now we only use git, but in the future we can extend this to use other downloaders e.g. OCI/...
+	var downloader downloadable
 	switch {
 	default:
-		downloader = newGitDownloader(r.tmpDir, schemaSpec, r.credentialResolver, credentialNSN)
+		downloader = newGitDownloader(r.tmpDir, schema, r.credentialResolver)
```

Comment:
Is it possible that we might have a mix of repository types/backends in the future? in this case I would loop over the spec and determine the correct downloader on a repository-basis, rather than a schema basis - what do you think?

Reply:
theoretically it is possible but I would do this once we add multiple providers. I am not sure how practical this is.